### PR TITLE
fix: Fixed incorrect comment form suggestion for too_long_first_doc_paragraph lint

### LINKED ISSUE #13309: `too_long_first_doc_paragraph` incorrect suggestion for module-level doc
### Summary

Related PR: #12993 

For `too_long_first_doc_paragraph` in module-level doc, it should suggest to add an empty line with `//!` under certain conditions. However, the current implementation will suggest to add `///` only.

The module-level doc is specified because I found this rule not checking crate-level doc, which may be another issue.

### Reproducer

I tried this code:

```rust
// in lib.rs
pub mod foo;

// in foo.rs
//! A very short summary.
//! A much longer explanation that goes into a lot more detail about
//! how the thing works, possibly with doclinks and so one,
//! and probably spanning a many rows. Blablabla, it needs to be over
//! 200 characters so I needed to write something longeeeeeeer.
```

I expected to see this happen:

```
warning: first doc comment paragraph is too long
 --> src/foo.rs:1:1
  |
1 | / //! A very short summary.
2 | | //! A much longer explanation that goes into a lot more detail about
3 | | //! how the thing works, possibly with doclinks and so one,
4 | | //! and probably spanning a many rows. Blablabla, it needs to be over
5 | | //! 200 characters so I needed to write something longeeeeeeer.
  | |_
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#too_long_first_doc_paragraph
  = note: `#[warn(clippy::too_long_first_doc_paragraph)]` on by default
help: add an empty line
  |
1 ~ //! A very short summary.
2 + //!
  |
```

Instead, this happened:

```
warning: first doc comment paragraph is too long
 --> src/foo.rs:1:1
  |
1 | / //! A very short summary.
2 | | //! A much longer explanation that goes into a lot more detail about
3 | | //! how the thing works, possibly with doclinks and so one,
4 | | //! and probably spanning a many rows. Blablabla, it needs to be over
5 | | //! 200 characters so I needed to write something longeeeeeeer.
  | |_
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#too_long_first_doc_paragraph
  = note: `#[warn(clippy::too_long_first_doc_paragraph)]` on by default
help: add an empty line
  |
1 ~ //! A very short summary.
2 + ///
  |
```

### Version

```text
rustc 1.82.0-nightly (c6db1ca3c 2024-08-25)
binary: rustc
commit-hash: c6db1ca3c93ad69692a4c4b5542f26fda4bf3aec
commit-date: 2024-08-25
host: x86_64-unknown-linux-gnu
release: 1.82.0-nightly
LLVM version: 19.1.0
```

### Additional Labels

_No response_

### PR DESCRIPTION
**Pull request by vHugoObject**
fixes #13309 
changelog: none

Comment form is now a variable and a new test for too_long_first_doc_paragraph was added.

### CODE REVIEW

File: tests/ui/too_long_first_doc_paragraph_module_level.fixed [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#![warn(clippy::too_long_first_doc_paragraph)]
+pub mod foo {
+
+    // in foo.rs
+    //! A very short summary.
+    //!
+//! A much longer explanation that goes into a lot more detail about
```
Comment (dswij) [inline tests/ui/too_long_first_doc_paragraph_module_level.fixed:7]: The suggestion has some spacing inconsistency here

Comment (vHugoObject) [reply tests/ui/too_long_first_doc_paragraph_module_level.fixed]: My bad. I added spacing after the comment form variable.

File: tests/ui/too_long_first_doc_paragraph_module_level.rs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#![warn(clippy::too_long_first_doc_paragraph)]
```
Comment (dswij) [inline tests/ui/too_long_first_doc_paragraph_module_level.rs:1]: Can we put this test in `tests/ui/too_long_first_doc_paragraph-fix.rs`?

**Review by dswij — COMMENTED** [review]
Thanks for this! Looks good, just a minor comment.

Can you squash some commits?

File: tests/ui/too_long_first_doc_paragraph_module_level.rs (hunk 2 above)
Comment (vHugoObject) [reply tests/ui/too_long_first_doc_paragraph_module_level.rs]: Yes

**Comment by vHugoObject** [comment]
> Thanks for this! Looks good, just a minor comment.
> 
> Can you squash some commits?

yes

**Review by dswij — COMMENTED** [review]
Looks great. Thanks for addressing the comment

**Pull request merged**